src/h5web/visualizations/heatmap/Mesh.tsx

```diff
@@ -12,28 +12,31 @@ interface Props {
 function Mesh(props: Props): JSX.Element {
   const { dims, textureData, axisOffsets } = props;
 
-  const { size } = useThree();
+  const { gl, size } = useThree();
   const { width, height } = size;
+  const [leftAxisWidth, bottomAxisHeight] = axisOffsets;
+
+  useFrame(() => {
+    // Set viewport of WebGL renderer to account for axis
+    gl.setViewport(
+      leftAxisWidth,
+      bottomAxisHeight,
+      width - leftAxisWidth,
+      height - bottomAxisHeight
+    );
+  });
```

Comment:
This is where the magic happens. With this call, all the computations relating to the camera can now be done as if the axis didn't exist. In other words, the "bounds" of the heatmap in camera space match the size of the canvas. This simplifies:
- the clamping computations in `usePanZoom()`
- the axis domains computations in `AxisGrid`
- the positioning and sizing of the mesh in `Mesh`.

Reply:
It also means that the axis don't have to have a white background, as nothing is rendered under them. Another interesting side effect is that the axis can now be dragged to pan the heatmap.

Reply:
:sparkles: I think the comment would be much clearer if you add the info that *all the computations relating to the camera can now be done as if the axis didn't exist.*

Reply:
I've updated the comment 💯